### ISSUE TITLE
Add a more concise box theme?

### ISSUE THREAD

**Issue by findmyway** [FedeClaudi/Term.jl#155]
![image](https://user-images.githubusercontent.com/5612003/188821861-d12cb566-f42e-4773-b7ce-5ef6babce378.png)

It seems there's still one blank line between rows.

(I guess with `box=:SIMPLE`, the border is just turned into invisible instead of being removed?)

**Comment by FedeClaudi** [comment]
Hi, 

Yeah, you're right, `SIMPLE` just has an empty line where other boxes would have a line character (see [here](https://github.com/FedeClaudi/Term.jl/blob/4ab195d85c274e7cc869c1dc5bbce39778b5b5b7/src/boxes.jl#L351)).
This is to keep consistency with other boxes as the layout shouldn't be affected by the box choice. As a consequence each cell has a height of three lines: box top, content, box bottom  (though top/bottom are shared between adjacent cells).

So currently this behavior is not a bug but by design, is it something you need to have a more compact layout?

Perhaps it's something that could be added in, though I imagine without lines and without spacing it can start to be difficult to distinguish table rows in some cases.

**Comment by findmyway** [comment]
> So currently this behavior is not a bug but by design,

I see, thanks!

> is it something you need to have a more compact layout?

Yep, I'm working on a little simulation game which uses Term.jl to render the state in the terminal. All looks pretty well! (By the way, the `Grid` is awesome!), the only little issue is that the space between rows is a bit too large.

<img width="692" alt="image" src="https://user-images.githubusercontent.com/5612003/188898053-7129c333-d291-4692-9c45-7492dbfc51d1.png">

I'm wondering that, is it possible to introduce another box theme, like `:COMPACT` or `:CONCISE` which removes the `top` and `bottom` in each cell?

https://github.com/FedeClaudi/Term.jl/blob/4ab195d85c274e7cc869c1dc5bbce39778b5b5b7/src/boxes.jl#L44-L54

Based on the current definition, it seems all the fields are required. I'm not very familiar with the code structure in Term.jl, my initial thought is to make `top` and `bottom` optional. Not sure whether doing so will break others or not.

**Comment by FedeClaudi** [comment]
Hey, 

First of all, THAT'S AWESOME. 
Please please let me know when you have this ready, I would love to play with it and I would like to use it as an example of exactly the kind of stuff I wish Term was used for. 

---
> By the way, the Grid is awesome!)

You should thank @t-bltg for that, he did a great work with it!


---
Regarding the proposed change, I don't think that would work. Objects using `Box` tend to assume that all components are there.  However, it turns out I had already implemented what you need, `Table` is a large unwieldy creature and it's hard to remember exactly how it all works...

But basically:
```julia
 Table(data; compact=true)
```

will remove spacing between the lines (before/after photos below)
<img width="184" alt="image" src="https://user-images.githubusercontent.com/17436313/188901294-71ebf6db-d602-4d21-8c54-f988bed07304.png">
<img width="236" alt="image" src="https://user-images.githubusercontent.com/17436313/188901315-7d9ad3ea-162e-4bcb-bb73-c9153cedab35.png">

Basically it's not printing the line between two consecutive rows. However there seems to be a bug: the first row still has the border underneath. I'm guessing [this](https://github.com/FedeClaudi/Term.jl/blob/4ab195d85c274e7cc869c1dc5bbce39778b5b5b7/src/tables.jl#L229) is the culprit as it sets `compact=false` for the first line if the header is being shown. It will require a bit of testing to make sure that's the problem and to fix it, I can try to do it later today, unless you want to have a go at it and send a PR?

**Comment by findmyway** [comment]
> Please please let me know when you have this ready, I would love to play with it and I would like to use it as an example of exactly the kind of stuff I wish Term was used for.

Will do!

 @t-bltg Great job! The `Grid` is very helpful when organizing different subplots.

I can confirm that, with `compact=false`, it just works! So I'm closing this issue right now.

I think another related but is the one below:

<img width="339" alt="image" src="https://user-images.githubusercontent.com/5612003/188905144-b3e3ee8f-a68e-4839-9e7d-7bcdc99f58b2.png">

There shouldn't be a line between the first row and the second row.
I tried to fix it but no luck...

**Comment by FedeClaudi** [comment]
> There shouldn't be a line between the first row and the second row

That's caused by the same bug: `compact=true` doesn't add a box line between rows, hence the more compact layout. for `box=SIMPLE`, when the bug adds the row under the first line you just see a space, for other boxes you see the actual line. 
If you set `show_header=false` it should disappear.
